src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -77,13 +84,70 @@ public void add(
         {
             throw new IllegalStateException("Too many realms");
         }
-        realmsIdsByName.put(realm, 1L << nextRealmBitShift++);
+        realmsIdsByName.put(realm, new OAuthRealm(1L << nextRealmBitShift++));
     }
 
     public long resolve(
-        String realm)
+        String realm,
+        String[] scopes)
```

Comment:
Change parameter names to `realmName` and `scopeNames`, then use `realm` instead of `realmObject` below.
Generally speaking, try to avoid including the type in the variable name.